sql/pre_install/tables.sql

```diff
@@ -100,13 +100,14 @@ CREATE TABLE _timescaledb_catalog.dimension (
   compress_interval_length bigint NULL,
   integer_now_func_schema name NULL,
   integer_now_func name NULL,
+  correlated boolean NOT NULL,
```

Comment:
Suggestion: Not sure what is the best solution here, but since we are adding this extra column anyway, I am wondering if it isn't best to add a `char` type with an explicit dimension type while we are changing this anyway. 

Currently, the dimension type is implicit from NULL:ness of other columns, but now with three types of dimensions it starts to get messy. Maybe we should just add an explicit type at this point since this is what the boolean is, but only of one type. If `correlated` is `true` we know this is a correlated dimension, but if it is `false` it could be any of the other two types. So why not make all types explicit here? I think it will also make the code easier to follow.

The only downside of doing it is that you need to backfill the type for existing entries, but that shouldn't be too much work since you are rewriting the table anyway.